src/builtins.c

```diff
@@ -755,7 +755,7 @@ DLLEXPORT int jl_substrtod(char *str, size_t offset, int len, double *out)
         pend = bstr+len;
     }
     *out = strtod_c(bstr, &p);
-    if ((p == bstr) || (p != pend) ||
+    if ((p == bstr) || (p != pend && !isspace((unsigned char)*p)) ||
```

Comment:
In the case of the original bug report, `p` would be the newline character, which we allow in jl_strtod